### DISCUSSION TITLE
Riverpod StreamProvider watch examples?

### DISCUSSION THREAD

**Discussion by alguintu** [isar/isar#1342] [Help]
Has anybody used Riverpod with StreamProvider to watch an object or collection? I can't seem to make it work or find any other examples.

Any leads would be greatly appreciated. Thanks.

**Comment by temcewen** [comment]
It's really easy. You just create the stream and then watch it from the Riverpod provider... 

Edit: corrected typo on Riverpod

**Comment by temcewen** [comment]
If you give a code sample of what you're trying that isn't working, maybe I can help you out more.

**Comment by alguintu** [comment]
After a few more hours of reading and trying, I found out I was missing the `fireImmediately` argument for the `watch` function that prevented me from seeing the result of the stream in the `ConsumerWidget`.

```
@riverpod
Stream<User?> user(UserRef ref) {
  return isar.users.watchObject(1, fireImmediately: true);
}
```

My stream works okay now, although I wonder how to properly use this without the fireImmediately flag. No matter what I did to the collection (crud), the stream didn't want to give values without it.

Any other important tips you all could give that are not immediately *obvious* I'd really appreciate it. Thanks!

**Reply by temcewen** [reply]
You shouldn't have to have the fireImmediately, but now that you mention it, I've actually never tried using a stream without fireImmediately enabled. It could be a bug, and I think it's more likely something wrong with Isar rather than Riverpod.

Also it needs to be noted that if you aren't actively using ref.watch to observe the userProvider, it's not going to be caching the most recently streamed value since the provider is set to autoDispose.

So I have 2 questions for you:

1. Your updates to the Isar collection are going through transactions, right?
2. Is userProvider being watched with "ref.watch" during the time that an Isar transaction is committed? Remember, ref.watch is different than ref.read.
